libs/algebra/include/nil/crypto3/algebra/curves/detail/forms/edwards/inverted/element_g1.hpp

```diff
@@ -97,6 +99,19 @@ namespace nil {
                             : X(X), Y(Y), Z(Z) 
                         { }
 
+                        /** @brief constructor from affine coordinates
+                         *
+                         */
+                        constexpr curve_element(const field_value_type& X, const field_value_type& Y)
+                            : X(X.inversed()), Y(Y.inversed()), Z(field_value_type::one())
+                        {
+                            /* Check whether constructed point has proper order */
+                            if ( *this * (scalar_field_type::modulus - 1) + *this != zero()) {
+                                throw std::runtime_error("Point has wrong order");
```

Comment:
Let's change the text to something like "EC points of order other than the group order are not allowed."